### PR TITLE
fixed camera example

### PR DESCRIPTION
**Pull request by nt4f04uNd**

### CODE REVIEW

**Comment by nt4f04uNd** [comment]
@acoutts can you review this?

**Comment by acoutts** [comment]
Sorry i've bene totally slammed, hoping to get to reviewing this later in the week.

File: packages/camera/camera/example/lib/main.dart [hunk 1]
```diff
@@ -607,38 +590,33 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
     if (controller != null) {
       await controller!.dispose();
     }
-    final CameraController cameraController = CameraController(
+    
+    final CameraController _controller = CameraController(
```
Comment (mvanbeusekom) [inline packages/camera/camera/example/lib/main.dart:594]: What would be the reason to rename this variable from `cameraController` to `_controller`? The original name was more descriptive and the added `_` would suggest it concerns a private class level field which is not the case.

File: packages/camera/camera/example/lib/main.dart [hunk 2]
```diff
@@ -886,31 +862,50 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
...
+        }
+        videoController!.removeListener(videoPlayerListener);
       }
-    };
-    vController.addListener(videoPlayerListener!);
-    await vController.setLooping(true);
-    await vController.initialize();
-    await videoController?.dispose();
+    }
+    final _videoController = VideoPlayerController.file(File(videoFile!.path));
```
Comment (mvanbeusekom) [inline packages/camera/camera/example/lib/main.dart:879]: Although the name is more semantic here, the `_` suggests the variable is a private class level field. The variable is a local method scope. Not sure if the semantic gains outweighs the scope. But maybe we can rename it something like `localVideoController`?

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Hi @nt4f04uNd, 

Could you submit this PR directly on the flutter/plugins repo? The pull request with the Android changes is already (too) large and reviewers are requesting us for finding ways to split it up into separate PRs. Adding these changes would only add to the collection of changed lines. Also your fixes for the example App would also add a lot of benefit while these changes are not yet merged in the main repo.

Besides that I added some addition questions / remarks to this PR which could maybe also taking into consideration when creating a separate PR.

File: packages/camera/camera/example/lib/main.dart [hunk 3]
```diff
@@ -27,8 +27,6 @@ IconData getCameraLensIcon(CameraLensDirection direction) {
       return Icons.camera_front;
     case CameraLensDirection.external:
       return Icons.camera;
-    default:
-      throw ArgumentError('Unknown lens direction');
```
[the remark this answers is not in the document]
Comment (nt4f04uNd) [reply packages/camera/camera/example/lib/main.dart:31]: some lint from nnbd was telling me about it
totally agree, will add it back

File: packages/camera/camera/example/lib/main.dart (hunk 2 above)
Comment (nt4f04uNd) [reply packages/camera/camera/example/lib/main.dart:879]: yes, makes sense

File: packages/camera/camera/example/lib/main.dart (hunk 1 above)
Comment (nt4f04uNd) [reply packages/camera/camera/example/lib/main.dart:594]: i renamed it to match the instance name `controller`, `_` was added to not overlap with it. `localController` better suits i guess

**Comment by nt4f04uNd** [comment]
i thought the progress on the original PR would be a little bit faster, but now i totally agree that i should make this as a separate PR

thanks for your comments, i will address them

**Pull request closed without merging**